ginuudan/kube.py

```diff
@@ -109,8 +122,33 @@ def port_forward(self, method, path, port):
         self.logger.debug(f"Response: {response.decode('utf-8')}")
         error = pf.error(port)
         if error is None:
-            self.logger.info(f"Successfully sent signal to {path}.")
+            self.event.create(
+                f"Successfully shutdown {container}",
+                self.namespace,
+                create_object_reference(),
+            )
         else:
             self.logger.error(f"Port {port} has the following error: {error}")
         # error can be used to return whether port forward succeeded,
         # but perhaps we should raise exceptions instead
+
+
+class Event:
+    def __init__(self, core_v1):
+        self.core_v1 = core_v1
+
+    def create(self, message, namespace, involved_object, action):
+        timestamp = datetime.now(timezone.utc)
+        event = core_v1.V1Event(
+            involved_object=involved_object,
+            message=message,
+            metadata=client.V1ObjectMeta(
+                name=f"ginuudan-{secrets.token_urlsafe(8)}",
+            ),
+            reason="ContainerShutdown",
+            source=client.V1EventSource(
+                component="ginuudan",
+            ),
+            type="Info",
```

Comment:
burde være "Warning" i stedet for info, syns du ikke? Ellers ser dette smooth ut!

Reply:
Kom på at vi burde gjøre event litt mer generell, slik at man sender inn `type` og `reason` kanskje? Eller lage en abstraksjon-funksjon?

Tok en titt på `events` i clusteret, og det ser ut som at `Normal` er riktig type. Den som blir brukt for å starte og stoppe containere i en pod. Ser at de bruker `reason` `Killing` også, så kanskje bytte til det.